### PR TITLE
fix: `//dd:span` on `func main()` does not work

### PR DESCRIPTION
**Pull request by RomainMuller**
Since the tracer was started at the start of the `main` function, a `//dd:span` annotation there may not have been honored, depending on the order in which aspects were woven into the file.

This PR mvoes the tracer initialization to a `func init()` function, so that it happens strictly before the `main` function starts executing, making it possible to `//dd:span` the `main` function reliably.

### CODE REVIEW

File: samples/client/main.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ import (
 	"time"
 )
 
+//dd:span
```
Comment (eliottness) [inline samples/client/main.go:18]: If I am not mistaken the samples tests do not test if traces are outputted. Wouldn't it be better to put this in the dd:span integration test ?

**Review by eliottness — APPROVED** [review]
Thoughts for later: would it be better to divide this aspect into 2: one for starting the tracer and one for the profiler ? Especially since we can add multiple init functions in the same file

File: samples/client/main.go (hunk 1 above)
Comment (RomainMuller) [reply samples/client/main.go:18]: We cannot have a `func main()` in tests... we could *maybe* do something by doing an `orchestrion go run` in there...

Comment (eliottness) [reply samples/client/main.go:18]: No no no. This is already an weird edge case, your explanation is enough

File: internal/injector/builtin/yaml/go-main.yml [hunk 2]
```diff
@@ -18,39 +18,55 @@ aspects:
...
+                  profiler.WithTags("orchestrion:true"),
+                )
+                if err != nil {
+                  // TODO: is there a better reporting mechanism?
+                  // The tracer and profiler already use the stdlib logger, so
+                  // we're not adding anything new. But users might be using a
+                  // different logger.
+                  log.Printf("failed to start profiling: %s", err)
+                }
+                defer profiler.Stop()
```
Comment (nsrip-dd) [inline internal/injector/builtin/yaml/go-main.yml:69]: This will stop the profiler as soon as `func init()` returns, meaning we won't get profiling. We could either keep the profiler start in `main`, or remove the call to `profiler.Stop`. I think either would work fine.

I still plan to set up some end-to-end testing of Orchestrion + profiling which would catch this, just had other priorities recently.

**Review by nsrip-dd — CHANGES_REQUESTED** [review]

File: internal/injector/builtin/yaml/go-main.yml (hunk 2 above)
Comment (RomainMuller) [reply internal/injector/builtin/yaml/go-main.yml:69]: Oh yikes, obviously! Good thing you're noticing this...

**Review by nsrip-dd — APPROVED** [review]
